sechub-schedule/src/main/java/com/daimler/sechub/domain/schedule/batch/SynchronSecHubJobExecutor.java

```diff
@@ -0,0 +1,158 @@
+import java.util.UUID;
+
+import org.jboss.logging.MDC;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.daimler.sechub.domain.schedule.ExecutionResult;
+import com.daimler.sechub.domain.schedule.job.ScheduleSecHubJob;
+import com.daimler.sechub.sharedkernel.LogConstants;
+import com.daimler.sechub.sharedkernel.messaging.BatchJobMessage;
+import com.daimler.sechub.sharedkernel.messaging.DomainMessage;
+import com.daimler.sechub.sharedkernel.messaging.DomainMessageService;
+import com.daimler.sechub.sharedkernel.messaging.DomainMessageSynchronousResult;
+import com.daimler.sechub.sharedkernel.messaging.IsSendingAsyncMessage;
+import com.daimler.sechub.sharedkernel.messaging.IsSendingSyncMessage;
+import com.daimler.sechub.sharedkernel.messaging.JobMessage;
+import com.daimler.sechub.sharedkernel.messaging.MessageDataKey;
+import com.daimler.sechub.sharedkernel.messaging.MessageDataKeys;
+import com.daimler.sechub.sharedkernel.messaging.MessageID;
+
+class SynchronSecHubJobExecutor {
+
+    private static final Logger LOG = LoggerFactory.getLogger(SynchronSecHubJobExecutor.class);
+    private DomainMessageService messageService;
+    private SecHubJobSafeUpdater secHubJobSafeUpdater;
+
+
+    public SynchronSecHubJobExecutor(DomainMessageService messageService, SecHubJobSafeUpdater secHubJobSafeUpdater) {
+        this.secHubJobSafeUpdater=secHubJobSafeUpdater;
+        this.messageService=messageService;
+    }
+    
+    @IsSendingSyncMessage(MessageID.START_SCAN)
+    public void execute(final ScheduleSecHubJob sechubJob,final Long batchJobId) {
+        Thread t = new Thread(new Runnable() {
+            
+            @Override
+            public void run() {
+                UUID secHubJobUUID = sechubJob.getUUID();
+                String secHubJobUUIDAsString= secHubJobUUID.toString();
+                
+                try {
+                    String secHubConfiguration = sechubJob.getJsonConfiguration();
+
+                    /* own thread so MDC.put necessary */
+                    MDC.clear();
+                    MDC.put(LogConstants.MDC_SECHUB_JOB_UUID, secHubJobUUIDAsString);
+                    MDC.put(LogConstants.MDC_SECHUB_PROJECT_ID, sechubJob.getProjectId());
+
+                    LOG.info("Executing sechub job: {}", secHubJobUUIDAsString);
+
+                    /* we send no a synchronous SCAN event */
+                    DomainMessage request = new DomainMessage(MessageID.START_SCAN);
+                    request.set(MessageDataKeys.EXECUTED_BY, sechubJob.getOwner());
+                    request.set(MessageDataKeys.SECHUB_UUID, secHubJobUUID);
+                    request.set(MessageDataKeys.SECHUB_CONFIG, MessageDataKeys.SECHUB_CONFIG.getProvider().get(secHubConfiguration));
+
+                    BatchJobMessage batchJobIdMessage = new BatchJobMessage();
+                    batchJobIdMessage.setBatchJobId(batchJobId);
+                    batchJobIdMessage.setSechubJobUUID(secHubJobUUID);
```

Comment:
```suggestion
                    batchJobIdMessage.setSecHubJobUUID(secHubJobUUID);
```

Reply:
Okay, I will change this, even when this is not inside the issue but a very old, existing method.

Reply:
done